src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -3321,6 +3328,26 @@ private void SetUpRPCParameters(_SqlRPC rpc, int startCount, bool inSchema, SqlP
                     // set default value bit
                     if (parameter.Direction != ParameterDirection.Output)
                     {
+                        // detect incorrectly derived type names unchanged by the caller and fix them
+                        if (parameter.IsDerivedParameterTypeName)
+                        {
+                            string[] parts = MultipartIdentifier.ParseMultipartIdentifier(parameter.TypeName, "[\"", "]\"", SR.SQL_TDSParserTableName, false);
+                            if (parts != null)
+                            {
+                                int length = parts.Length;
+                                if (
+                                    length > 3 && //require at least 3 parts
+                                    parts[length - 1] != null && // name must not be null
+                                    parts[length - 2] != null && // schema must not be null
+                                    parts[length - 3] != null && // server should not be null or we don't need to remove it
+                                    (length == 3 || parts[length - 4] == null) // if a database space is availble then is must be null
```

Comment:
If length == 3, then it won't pass the if block as the first check says Length has to be > 3. I guess it would be >=3 at the beginning?

Reply:
I agree and will push the change but I can't rebuild and test at the moment (build issue)